lib/components/WalletContextProvider.jsx

```diff
@@ -148,11 +148,21 @@ const doConnectWallet = async (walletType, setOnboardState) => {
 const connectWallet = (w, setOnboardState) => {
   Cookies.set(SELECTED_WALLET_COOKIE_KEY, w.name, cookieOptions)
 
+  const provider = new ethers.providers.Web3Provider(w.provider, 'any')
+  provider.on('network', (newNetwork, oldNetwork) => {
+    // When a Provider makes its initial connection, it emits a "network"
+    // event with a null oldNetwork along with the newNetwork. So, if the
+    // oldNetwork exists, it represents a changing network
+    if (oldNetwork) {
+      window.location.reload()
```

Comment:
We don't need this full refresh since the top level hooks are re-fetching when the `chainId` updates. This just  slows down the UI between switching networks since everything needs to be re-fetched and re-rendered. It also blows away any caching of network A pools & pool data when switching to network B.

Reply:
I agree, the refresh sucks. I was following the suggestion on the ethers issue and didn't test how it would work without the refresh. The problem is that when you try to make a transaction (deposit, withdraw, etc) after changing the network it would fail. It may be enough to just have `'any'` as the 2nd param.